src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -71,46 +72,48 @@ private void HandleClusterDatabaseChanged((string DatabaseName, long Index, stri
                     return;
                 }
 
-                if (record.DeletionInProgress != null &&
-                    record.DeletionInProgress.TryGetValue(_serverStore.NodeTag, out DeletionInProgressStatus deletionInProgress) &&
-                    deletionInProgress != DeletionInProgressStatus.No)
+                var deletionInProgress = DeletionInProgressStatus.No;
+                var directDelete = record.DeletionInProgress != null &&
+                                   record.DeletionInProgress.TryGetValue(_serverStore.NodeTag, out deletionInProgress) &&
+                                   deletionInProgress != DeletionInProgressStatus.No;
+
+                string mentorsChangeVector = "";
+                var conditionalDelete = record.DeletionInProgressChangeVector != null &&
+                                    record.DeletionInProgressChangeVector.TryGetValue(_serverStore.NodeTag, out mentorsChangeVector);
+
+                if (directDelete)
                 {
-                    UnloadDatabase(t.DatabaseName);
+                    DeleteDatabase(t, deletionInProgress, record);
+                    return;
+                }
 
-                    if (deletionInProgress == DeletionInProgressStatus.HardDelete)
-                    {
-                        RavenConfiguration configuration;
-                        try
-                        {
-                            configuration = CreateDatabaseConfiguration(t.DatabaseName, ignoreDisabledDatabase: true, ignoreBeenDeleted: true, databaseRecord: record);
-                        }
-                        catch (Exception ex)
-                        {
-                            configuration = null;
-                            if (_logger.IsInfoEnabled)
-                                _logger.Info("Could not create database configuration", ex);
-                        }
-                        // this can happen if the database record was already deleted
-                        if (configuration != null)
-                        {
-                            DatabaseHelper.DeleteDatabaseFiles(configuration);
-                        }
+                if (conditionalDelete)
+                {
+                    // A delete command was issued while we were in rehabilitation. We need to check if we recieved any new documents while we were in that state.
+                    // If so, we can't simply delete the database.
 
-                        // At this point the db record still exists but the db was effectively deleted 
-                        // from this node so we can also remove its secret key from this node.
-                        if (record.Encrypted)
+                    DatabasesCache.TryGetValue(t.DatabaseName, out var dbTask);
+                    var db = dbTask.Result;
```

Comment:
Why promise do you have that this is already loaded and you aren't blocking the raft processing thread?